traefik/templates/deployment.yaml

```diff
@@ -49,6 +49,10 @@ spec:
         app.kubernetes.io/managed-by: {{ .Release.Service }}
         app.kubernetes.io/instance: {{ .Release.Name }}
     spec:
+      {{- if .Values.deployment.imagePullSecrets }}
+      imagePullSecrets:
+{{ toYaml .Values.deployment.imagePullSecrets | indent 6 }}
```

Comment:
That indentation looks off?

Reply:
@SantoDE Thanks for the review!

The indent function of helm starts at the first character of the value so it has to be either:
```
    spec:
      {{- if .Values.deployment.imagePullSecrets }}
      imagePullSecrets:
{{ toYaml .Values.deployment.imagePullSecrets | indent 6 }}
      {{- end }}
```
or without indent function:
```
    spec:
      {{- if .Values.deployment.imagePullSecrets }}
      imagePullSecrets:
      {{ toYaml .Values.deployment.imagePullSecrets }}
      {{- end }}
```
To get this result:
```
    spec:
      imagePullSecrets:
      - name: myRegistryKeySecretName
```